src/modules/oAuth/oAuth.service.ts

```diff
@@ -0,0 +1,41 @@
+import { Injectable } from '@nestjs/common';
+import { UsersRepository } from '../users/users.repository';
+import { generateToken } from 'src/shared';
+
+@Injectable()
+export class OAuthService {
+  constructor(readonly userRepository: UsersRepository) {}
+  async googleLogin(req) {
+    const isUserExist = await this.userRepository.getUserBy({
```

Comment:
I think it is okay to name the variable "isUserExist" if you use that variable only inside the if condition, but you are also using it for other things. I think just "user' is a better option in this case.